src/databricks/labs/ucx/source_code/base.py

```diff
@@ -54,6 +56,14 @@ def as_deprecation(self) -> 'Deprecation':
     def as_convention(self) -> 'Convention':
         return Convention(**self.__dict__)
 
+    def for_path(self, path: Path) -> LocatedAdvice:
+        return LocatedAdvice(path=path, **self.__dict__)
+
+
+@dataclass
+class LocatedAdvice(Advice):
```

Comment:
```suggestion
class LocatedAdvice:
    advice: Advice
```
i think this will make it more clear

Reply:
done